tests/Rolle.spec.ts

```diff
@@ -1,193 +1,293 @@
+      await expect(rolleCreationConfirmPage.label_Angebote).toHaveText(
+        "Zugeordnete Angebote:",
+      );
+      await expect(rolleCreationConfirmPage.data_Angebote).toContainText(
+        AngebotA,
+      );
+      await expect(rolleCreationConfirmPage.data_Angebote).toContainText(
+        AngebotB,
+      );
+      await expect(rolleCreationConfirmPage.data_Angebote).toContainText(
+        AngebotC,
+      );
+      await expect(rolleCreationConfirmPage.label_Systemrechte).toHaveText(
+        "Systemrechte:",
+      );
+      await expect(rolleCreationConfirmPage.data_Systemrechte).toContainText(
+        SystemrechtA + ", " + SystemrechtB + ", " + SystemrechtC,
+      );
+      await expect(
+        rolleCreationConfirmPage.button_WeitereRolleAnlegen,
+      ).toBeVisible();
+      await expect(
+        rolleCreationConfirmPage.button_ZurueckErgebnisliste,
+      ).toBeVisible();
     });
 
     await test.step(`Rolle wieder löschen`, async () => {
-      const RollenID = await getRolleId(page, ROLLENNAME); 
+      const RollenID = await getRolleId(page, ROLLENNAME);
       await deleteRolle(page, RollenID);
     });
   });
-});
\ No newline at end of file
+});
+
+test.describe("Testet die Anlage einer neuen Rolle", () => {
+  let roleName: string | undefined = undefined;
+
+  test("Eine neue Rolle anlegen und sicherstellen, dass alle Serviceprovider angezeigt werden und verfügbar sind @long", async () => {
+    const rolleCreationView: RolleCreationViewPage =
+      await test.step("Rolle anlegen aufrufen", async () => {
+        return await startseite
+          .administration()
+          .then((menu) => menu.rolleAnlegen());
+      });
+
+    const rolleCreationConfirm: {
+      rolleCreationConfirmPage: RolleCreationConfirmPage;
+      selectedSPs: string[];
+    } = await test.step("Rolle mit mehr als 5 SPs anlegen", async () => {
+      await rolleCreationView.schulstrukturknoten.selectByTitle(
+        "Land Schleswig-Holstein",
+      );
+      await rolleCreationView.rollenarten.selectByTitle("Lehr");
+      roleName = "Neue Rolle aus Test";
+      await rolleCreationView.enterRollenname(roleName);
+      const theFirstSeven = Array.from({ length: 7 }, (_, key) => key);
+      const selectedItems: string[] =
+        await rolleCreationView.angebote.selectByPosition(theFirstSeven);
+      return {
+        rolleCreationConfirmPage: await rolleCreationView.createRolle(),
+        selectedSPs: selectedItems,
+      };
+    });
+
+    const rolleManagementPage =
+      await test.step("Anlage prüfen und zurück", async () => {
+        const { rolleCreationConfirmPage } = rolleCreationConfirm;
+        await expect(
+          rolleCreationConfirmPage.confirmationMessage,
+        ).toBeVisible();
+        return rolleCreationConfirmPage.backToResultList();
+      });
+
+    await test.step("Rollentabelle prüfen", async () => {
+      const row = rolleManagementPage.rowByRoleName(roleName);
```

Comment:
In VS Code wird roleName als Error markiert: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.ts(2345)